### ISSUE TITLE
Does Traefik support Swarm?

### ISSUE THREAD

**Issue by jakobrosenberg** [containous/traefik#143]
I tried spinning up a master and a few agents with the virtualbox driver. Traefik seems to only see the containers running locally on master.

Am I doing something wrong or is swarm not supported?

**Comment by emilevauge** [comment]
Hi @jakobrosenberg, it seems to be a bug. But it should work with Swarm as it is the same API. Could you provide traefik `debug` logs while starting ?

**Comment by vdemeester** [comment]
@jakobrosenberg I'm also working on adding integration test on swarm to make sure it works and keep working, so logs/repro./… would be super useful :wink:

**Comment by jakobrosenberg** [comment]
To my recollection I followed the instructions on this page. https://docs.docker.com/swarm/install-w-machine

this is my docker-compose.yml
```
web:
    build: . 
    labels:
     - "traefik.port=80"
     - "traefik.backend=mysite"
     - "traefik.frontend.passHostHeader=true"
     - "traefik.frontend.rule=Host"
     - "traefik.frontend.value=mysite.com"
    restart: always
```

The container only shows on the master through docker-machine and the --swarm parameter. ``docker-machine env mysite --swarm``

**Comment by BrianAdams** [comment]
I can see all of the containers across the swarm when using the swarm tcp socket and the tls certificates.  The only issue I see is that traefik is registering the bridge IPs when it should be registering the public IPs.  This is a problem since in a multi-host swarm, the processes on one host don't have routes to the bridge networks of the other hosts.

**Comment by navinSing** [comment]
@BrianAdams 
>> The only issue I see is that traefik is registering the bridge IPs
Even then routing should happen .. right ..what was the issue when one container was trying to connect to another ?

**Comment by BrianAdams** [comment]
@navinSing, with only the bridge IP being registered, an instance of traefik can only route traffic to the same swarm-node it happens to be deployed to.  To be useful in a swarm, it needs a per container configuration option to let it know which network adapter/network to forward the traffic to on the container.  If that exists, you can then choose 'host' to use the public IP assigned by docker, or you could use 'overlay1' to reference an overlay network that both the traefik and the container have been connected to and it could successfully proxy to a container on any swarm-node.

**Comment by emilevauge** [comment]
Fixed by #313
